### PR TITLE
[bugfix] support endless polls, and misskey's' method of inferring expiry in closed polls

### PR DESCRIPTION
**Pull request by NyaaaWhatsUpDoc**
# Description

As the title says really. Misskey doesn't always set the `endTime` AS property, both in the case of an endless poll (where `closed` is also unset), or in the case that an expiring poll is now closed and it uses the set `closed` property to infer the past expiry. As such there's a required database migration in order to support empty `endTime` property.

## Checklist
- [x] I/we have read the [GoToSocial contribution guidelines](https://github.com/superseriousbusiness/gotosocial/blob/main/CONTRIBUTING.md).
- [x] I/we have discussed the proposed changes already, either in an issue on the repository, or in the Matrix chat.
- [x] I/we have performed a self-review of added code.
- [x] I/we have written code that is legible and maintainable by others.
- [x] I/we have commented the added code, particularly in hard-to-understand areas.
- [ ] I/we have made any necessary changes to documentation.
- [x] I/we have added tests that cover new code.
- [x] I/we have run tests and they pass locally with the changes.
- [x] I/we have run `go fmt ./...` and `golangci-lint run`.

### CODE REVIEW

File: internal/federation/dereferencing/util.go [hunk 1]
```diff
@@ -17,6 +17,12 @@
 
 package dereferencing
 
+import (
+	"slices"
```
Comment (tsmethurst) [inline internal/federation/dereferencing/util.go:21]: Yay new slices functions :)

Comment (NyaaaWhatsUpDoc) [reply internal/federation/dereferencing/util.go:21]: the slices package is great!

File: internal/db/bundb/migrations/20231110142330_small_poll_table_tweaks.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+package migrations
+
+import (
+	"context"
+
+	"github.com/uptrace/bun"
+)
+
+func init() {
+	up := func(ctx context.Context, db *bun.DB) error {
```
[the remark this answers is not in the document]
Comment (NyaaaWhatsUpDoc) [reply internal/db/bundb/migrations/20231110142330_small_poll_table_tweaks.go:27]: oh that is some interesting behaviour ... shall do!

[the remark this answers is not in the document]
Comment (NyaaaWhatsUpDoc) [reply internal/db/bundb/migrations/20231110142330_small_poll_table_tweaks.go:27]: i just checked myself in postgres and adding a new column with different constraint, dropping old and renaming the new doesn't maintain the constraint

[the remark this answers is not in the document]
Comment (tsmethurst) [reply internal/db/bundb/migrations/20231110142330_small_poll_table_tweaks.go:27]: Alright cool :) Probably that's just for indexes then.

**Pull request merged**